spec/languages/lambda/lambda-parser-test.js

```diff
@@ -0,0 +1,154 @@
+{ type: "lambda",
+  vars: [ "x" ],
+  body: { type: "num", value: 10 }
+},
+  {
+    "type": "call",
+    "func": { "type": "var", "value": "foo" },
+    "args": [
+    { "type": "var", "value": "a" },
+    { "type": "num", "value": 1 }
+    ]
+  },
+  {
+    "type": "if",
+    "cond": { "type": "var", "value": "foo" },
+    "then": { "type": "var", "value": "bar" },
+    "else": { "type": "var", "value": "baz" }
+  },
+  {
+    "type": "if",
+    "cond": { "type": "var", "value": "foo" },
+    "then": { "type": "var", "value": "bar" }
+  },
+  {
+    "type": "assign",
+    "operator": "=",
+    "left": { "type": "var", "value": "a" },
+    "right": { "type": "num", "value": 10 }
+  },
+  {
+    "type": "binary",
+    "operator": "+",
+    "left": { "type": "var", "value": "x" },
+    "right": {
+      "type": "binary",
+      "operator": "*",
+      "left": { "type": "var", "value": "y" },
+      "right": { "type": "var", "value": "z" }
+    }
+  },
+  {
+    "type": "prog",
+    "prog": [
+      {
+        "type": "assign",
+        "operator": "=",
+        "left": { "type": "var", "value": "a" },
+        "right": { "type": "num", "value": 5 }
+      },
+      {
+        "type": "assign",
+        "operator": "=",
+        "left": { "type": "var", "value": "b" },
+        "right": {
+          "type": "binary",
+          "operator": "*",
+          "left": { "type": "var", "value": "a" },
+          "right": { "type": "num", "value": 2 }
+        }
+      },
+      {
+        "type": "binary",
+        "operator": "+",
+        "left": { "type": "var", "value": "a" },
+        "right": { "type": "var", "value": "b" }
+      }
+    ]
+  },
+  {
+    "type": "let",
+    "vars": [
+      {
+        "name": "a",
+        "def": { "type": "num", "value": 10 }
+      },
+      {
+        "name": "b",
+        "def": {
+          "type": "binary",
+          "operator": "*",
+          "left": { "type": "var", "value": "a" },
+          "right": { "type": "num", "value": 10 }
+        }
+      }
+    ],
+    "body": {
+      "type": "binary",
+      "operator": "+",
+      "left": { "type": "var", "value": "a" },
+      "right": { "type": "var", "value": "b" }
+    }
+  }];
+
+describe("test suite", function() {
+
+  var parserAst;
+  var correctAst;
+
+  for (var i = 0; i < codeUnits.length; i++) {
+
+    parserAst = main(codeUnits[i]);
+    correctAst = astUnits[i];
+
+    console.log(parserAst); //test
+    console.log(correctAst);
+
+    it("tests" + " " + codeUnits[i], function() {
+      expect(parserAst).toEqual(correctAst);
+    });
+  }
+});
+
+xdescribe("single test", function() { //for debugging purposes
+  it("123.5 num test", function() {
+    expect(main('123.5')).toEqual({ 'type': "num", value: 123.5 });
+  });
+});
+
+xdescribe("testing tests", function() {
+  it("for true equals true", function() {
+    expect(true).toBe(true);
+  });
+  it("for false equals true", function() {
+    expect(false).toBe(true);
+  });
+});
```

Comment:
if you feel like this is done, you should ditch these x'd out tests and the `console.log` statements.